packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -422,51 +416,387 @@ describe('Appointment Popup Form', () => {
+      const hasSettingsButtonRecurrent = buttonsRecurrent?.some((btn) => btn.name === 'settings');
 
-      jest.useFakeTimers();
+      expect(repeatEditorRecurrent.option('value')).toBe('daily');
+      expect(hasSettingsButtonRecurrent).toBe(true);
 
-      POM.popup.getCancelButton().click();
-
-      jest.runAllTimers();
+      scheduler.hideAppointmentPopup();
 
       POM.openPopupByDblClick('common-app');
 
-      expect(POM.popup.getSwitchByName('repeat').value).toBe('false');
-      expect(POM.popup.getInputByLabel('Subject').value).toBe('common-app');
+      // @ts-expect-error
+      const repeatEditorCommon = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      const buttonsCommon = repeatEditorCommon.option('buttons');
+      const hasSettingsButtonCommon = buttonsCommon?.some((btn) => btn.name === 'settings');
+
+      expect(repeatEditorCommon.option('value')).toBe('never');
+      expect(hasSettingsButtonCommon).toBe(false);
     });
 
-    it.skip('Recurrence repeat-end editor should have default \'never\' value after reopening appointment popup', async () => {
+    it('Recurrence repeat-end editor should have default \'never\' value after reopening appointment popup', async () => {
       const firstAppointment = { startDate: new Date(2015, 1, 9), endDate: new Date(2015, 1, 9, 1), text: 'caption 1' };
       const secondAppointment = { startDate: new Date(2015, 1, 9), endDate: new Date(2015, 1, 9, 1), text: 'caption 2' };
       const { POM, scheduler } = await createScheduler(getDefaultConfig());
 
       scheduler.showAppointmentPopup(firstAppointment);
-      POM.popup.getSwitchByName('repeat').click();
 
-      POM.popup.selectRadio('After');
+      // @ts-expect-error
+      const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor.option('value', 'weekly');
 
-      POM.popup.getSaveButton().click();
+      scheduler.hideAppointmentPopup();
 
       scheduler.showAppointmentPopup(secondAppointment);
-      POM.popup.getSwitchByName('repeat').click();
 
-      const radioValue = POM.popup.getSelectedRadioValue();
+      // @ts-expect-error
+      const repeatEditorSecond = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      const value = repeatEditorSecond.option('value');
+
+      expect(value).toBe('never');
+    });
+
+    it('Recurrence editor container should be visible after changing its visibility value', async () => {
+      const appointment = {
+        text: 'Test Appointment',
+        startDate: new Date(2017, 4, 1, 9, 30),
+        endDate: new Date(2017, 4, 1, 11),
+      };
+
+      const { POM, scheduler } = await createScheduler(getDefaultConfig());
+
+      scheduler.showAppointmentPopup(appointment);
+
+      const recurrenceGroup = $(POM.popup.element).find('.dx-scheduler-form-recurrence-group');
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(true);
```

Comment:
please add a function to popup's POM that will return 'main' or 'recurrence' depending on the current opened form.